claat/parser/gdoc/parse.go

```diff
@@ -110,23 +110,31 @@ const (
 )
 
 type docState struct {
-	clab     *types.Codelab // codelab and its metadata
-	totdur   time.Duration  // total codelab duration
-	survey   int            // last used survey ID
-	css      cssStyle       // styles of the doc
-	step     *types.Step    // current codelab step
-	lastNode types.Node     // last appended node
-	env      []string       // current enviornment
-	cur      *html.Node     // current HTML node
-	flags    stateFlag      // current flags
-	stack    []*stackItem   // cur and flags stack
+	clab         *types.Codelab  // codelab and its metadata
+	totdur       time.Duration   // total codelab duration
+	survey       int             // last used survey ID
+	css          cssStyle        // styles of the doc
+	step         *types.Step     // current codelab step
+	lastNode     types.Node      // last appended node
+	env          []string        // current enviornment
+	cur          *html.Node      // current HTML node
+	flags        stateFlag       // current flags
+	stack        []*stackItem    // cur and flags stack
+	passMetadata map[string]bool // set of metadata fields to pass along.
 }
 
 type stackItem struct {
 	cur   *html.Node
 	flags stateFlag
 }
 
+func newDocState() *docState {
+	ds := new(docState)
```

Comment:
From what I've read the `new` keyword is discouraged in favor of struct declarations.

Any special rational for this here since it doesn't seem to be doing anything special?

Reply:
I'd also make this method take in a variadic/optional arg for `css` since it seems to be a common assignment right after.

Reply:
Replaced with struct init